src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/ApplicationInsightsEventListener.kt

```diff
@@ -14,34 +17,38 @@ class ApplicationInsightsEventListener(
   private val authenticationFacade: AuthenticationFacade,
   private val telemetryClient: TelemetryClient,
 ) : VideoLinkBookingEventListener {
-  override fun bookingCreated(booking: VideoLinkBooking, specification: VideoLinkBookingSpecification, agencyId: String) {
+  override fun bookingCreated(
+    booking: VideoLinkBooking,
+    specification: VideoLinkBookingSpecification,
+    agencyId: String
+  ) {
     val properties = mutableMapOf(
       "id" to (booking.id?.toString()),
-      "bookingId" to booking.main.bookingId.toString(),
+      "bookingId" to booking.offenderBookingId.toString(),
       "court" to specification.court,
       "user" to authenticationFacade.currentUsername,
       "agencyId" to agencyId,
       "madeByTheCourt" to specification.madeByTheCourt.toString(),
     )
 
-    properties.putAll(appointmentDetail(booking.main, specification.main))
-    booking.pre?.also { properties.putAll(appointmentDetail(it, specification.pre!!)) }
-    booking.post?.also { properties.putAll(appointmentDetail(it, specification.post!!)) }
+    booking.appointments[MAIN]?.also { properties.putAll(appointmentDetail(it, specification.main)) }
```

Comment:
guess could add methods on the booking to make these look like fields:

`main get() = appointments[MAIN]`

Reply:
var mainAppointment: VideoLinkAppointment